### PR TITLE
Warn for CRS inaccuracies

### LINKED ISSUE #6706: Show a warning when CRSes based on a datum ensemble are selected (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#43088
Author: @nyalldawson
QGIS version: 3.20

**Show a warning when CRSes based on a datum ensemble are selected**

### PR Description:
Shows a warning in the projection selection widget when a CRS based on a datum ensemble is selected, warning the user that
there's an inherent lack of accuracy in the selected CRS

Requires PROJ 8+

https://user-images.githubusercontent.com/1829991/117229831-1efdae80-ae5f-11eb-8b18-8158aa97ebef.mp4

### Commits tagged with [need-docs] or [FEATURE]
"\n\n\n\n Show a warning in the projection selection widget when\na CRS based on a datum ensemble is selected, warning the user that\nthere's an inherent lack of accuracy in the selected CRS\n\nRequires PROJ 8+"

### LINKED ISSUE #6708: [feature] Add optional (not on by default) layer tree warning icon for layers with CRS inaccuracies (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#43201
Author: @nyalldawson
QGIS version: 3.20

**[feature] Add optional (not on by default) layer tree warning icon for layers with CRS inaccuracies**

### PR Description:
If the new "Show CRS accuracy warnings for layers in project legend"
is checked by a user, then any layers with a CRS with accuracy
issues (i.e. a dynamic crs with no coordinate epoch available,
or a crs based on a datum ensemble with accuracy exceeding the
user-set limit) will have a new warning icon reflecting that
the layer is a low-accuracy layer.

This is entirely opt-in, and designed for use in engineering/BIM/...
industries where inaccuracies of meter/submeter level are very
dangerous.

### Commits tagged with [need-docs] or [FEATURE]
"[feature] Add optional (not on by default) layer tree warning icon\nfor layers with CRS inaccuracies\n\nIf the new \"Show CRS accuracy warnings for layers in project legend\"\nis checked by a user, then any layers with a CRS with accuracy\nissues (i.e. a dynamic crs with no coordinate epoch available,\nor a crs based on a datum ensemble with accuracy exceeding the\nuser-set limit) will have a new warning icon reflecting that\nthe layer is a low-accuracy layer.\n\nThis is entirely opt-in, and designed for use in engineering/BIM/...\nindustries where inaccuracies of meter/submeter level are very\ndangerous."

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #6706, fixes #6708
I think I got the features but it may be interesting to provide more education regarding datum ensemble/crs static vs dynamic, inaccuracies... (and I'm not enough skilled for this).
@nyalldawson may I ask for your help/review, please? Thanks

### CODE REVIEW

File: docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -307,6 +307,8 @@ symbols are:
   close this project. To avoid data loss and make the layer permanent, click
   the icon to store the layer in any of the OGR vector formats supported by QGIS.
 * |indicatorNoCRS| to identify a layer that has no/unknown CRS
+* |indicatorLowAccuracy| for layers with crs inaccuracies (requires the
```
Comment (nyalldawson) [inline docs/user_manual/introduction/general_tools.rst:310]:
```suggestion
* |indicatorLowAccuracy| for layers with coordinates stored in a coordinate reference system which is inherently low accuracy (requires the
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 2]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
 
+.. _crs_inaccuracies:
+
+**Accuracy warnings**
+
+.. A small intro either on accuracy differences between CRS/datum,
+   or static vs dynamic CRS would be nice here.
+   Or if you want, you can expand a lot in the working_with_projection.rst file.
+
+   Also if anyone knows a link to "datum ensemble" concept?
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:264]: Unfortunately not. There's various blog posts which are decent summaries, but nothing formal enough to include in the documentation. (I'd love someone to petition osgeo to fund a well written, peer reviewed intro-level explanation to these concepts)

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 3]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
+
+:guilabel:`Only show warnings for CRS inaccuracies which exceed` a given
+distance: occurs when you are explicitly creating or modifying a data and
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:267]:
```suggestion
distance: occurs when you are explicitly creating or modifying a dataset and
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 4]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
+select in the CRS selector widget a CRS based on a datum ensemble with lower
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:268]:
```suggestion
select a CRS based on a datum ensemble with lower
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 5]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
+accuracy. The default is to ``Always show`` the warning if any inaccuracy.
+Requires a QGIS version using at least `PROJ 8.0 <https://proj.org/index.html>`_.
+
+|unchecked| :guilabel:`Show warning for CRS inaccuracies for layers in project legend`:
+If checked, any layer with a CRS with accuracy issues (i.e. a dynamic crs with
+no coordinate epoch available, or a crs based on a datum ensemble with accuracy
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:274]:
```suggestion
no coordinate epoch available, or a CRS based on a datum ensemble with inherent inaccuracy
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 6]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
+exceeding the user-set limit) will have the |indicatorLowAccuracy| warning icon
+in the :guilabel:`Layers` panel reflecting that it is a low-accuracy layer.
+
+This is designed for use in engineering, BIM,... industries where inaccuracies
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:278]:
```suggestion
This is designed for use in engineering, BIM, asset management, and other fields where inaccuracies
```

File: docs/user_manual/introduction/qgis_configuration.rst [hunk 7]
```diff
@@ -253,6 +253,31 @@ or when a layer without a CRS is loaded.
...
+of meter/submeter level are very dangerous.
```
Comment (nyalldawson) [inline docs/user_manual/introduction/qgis_configuration.rst:279]:
```suggestion
of meter/submeter level are potentially very dangerous or expensive!
```

**Comment by nyalldawson** [comment]
What you've got looks good :+1: . I agree that we are missing a well written beginner level explanation of these concepts, but as noted above I think it's outside of the scope of QGIS to do this. I'd rather see osgeo fund a professional writer, advised by the proj team, to create generic, professional level documents for use explaining this.

**Comment by DelazJ** [comment]
Thanks Nyall for the quick review.
About osgeo writing on the concepts, I saw your previous "call" in one of the PR discussion and I agree that a doc for reference is needed. Could this request be raised by our PSC? (I have no idea what is the osgeo procedure)

**Pull request merged**